staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -245,9 +247,13 @@ func (config *DirectClientConfig) getUserIdentificationPartialConfig(configAuthI
 	}
 	if len(configAuthInfo.ClientCertificate) > 0 || len(configAuthInfo.ClientCertificateData) > 0 {
 		mergedConfig.CertFile = configAuthInfo.ClientCertificate
-		mergedConfig.CertData = configAuthInfo.ClientCertificateData
+		if len(mergedConfig.CertFile) == 0 {
+			mergedConfig.CertData = configAuthInfo.ClientCertificateData
```

Comment:
same comment here... I don't think this is the correct place to do conditional copying, and this reverses the documented precedence of ClientCertificateData overriding ClientCertificate